packages/AI/Mistral/src/models/mistralEmbedding.ts

```diff
@@ -15,13 +15,43 @@ export class MistralEmbedding extends Embeddings {
         }
     }
 
-    public async EmbedText(params: EmbedParams): Promise<EmbedResult> {
-        const response: EmbeddingResponse = await MistralEmbedding._client.embeddings(params.model, params.text);
+    public get Client(): MistralClient { return MistralEmbedding._client; }
+
+    /**
+     * Mistral AI embedding endpoint outputs vectors in 1024 dimensions
+     */
+    public async EmbedText(params: EmbedTextParams): Promise<EmbedTextResult> {
+        params.model = params.model || "mistral-embed";
+        const response: EmbeddingResponse = await MistralEmbedding._client.embeddings(params.model, [params.text]);
+        return {
+            object: response.object,
+            model: response.model,
+            ModelUsage: new ModelUsage(response.usage.prompt_tokens, response.usage.completion_tokens),
+            vector: response.data[0].embedding
```

Comment:
Heads up that `response.data` could be an empty array, which would throw an error when trying to access `.embedding` on undefined. The Typescript compiler can't know this ahead of time, unfortunately